feat(mobile): review remote-trashed assets before syncing deletions on Android (part #1)

PeterOmbodi opened the pull request:
## Description

_**Part 1 of the flow: Determining and displaying assets for trash sync**_

An additional synchronization mode for handling changes between the remote trash and the Android device.
It allows users to review these changes either on a dedicated Review screen or directly when viewing an asset from the main timeline.
Available actions are: keep the asset on the device (reject the remote deletion) or move it to the local trash (accept the change).
When an asset is restored from the cloud trash, the app attempts to restore it locally in automatic mode.

## How Has This Been Tested?

To verify this, you need an Android 11+ device.

1. Launch the Flutter app on an Android device.
2. Log in to the app (I used the demo server and demo account).
3. Go to Settings → Advanced and enable the Review remote deletions switch.
4. Grant Manage Media Access if it has not been granted yet.
5. Include an album in the device’s backup list.
6. Download any asset (or multiple assets) from the backed-up album.
7. Open the Immich web app and log in with the same account (I used the demo).
8. In the web app, delete (move to trash) an asset that is downloaded locally on your phone (marked with the checked cloud icon).
9. If your device uses Google Photos as the gallery app, make sure syncing is disabled or paused - otherwise, a file moved to trash may appear both in Trash and Photos.
10. Return to the mobile app.
11. Tap the user profile icon. The Review out-of-sync changes action should appear with a count of unsynced items - tap it.
12. On the review screen, select a group and choose Keep or Delete.
13. On the review screen, open a preview and choose Keep or Delete.

## Screenshots

| **p.3** – Review remote deletions switch | **p.11** - Out-of-sync changes action button - Dark theme | **p.11** - Out-of-sync changes action button - Light theme |
|---|---|---|
|<img width="340" alt="image" src="https://github.com/user-attachments/assets/6ff1c491-3b73-4e8c-98f6-3092c87123fe" />|<img width="270" alt="image" src="https://github.com/user-attachments/assets/e5e3e3c5-bbbf-46ba-8d1f-0e02795c1853" />|<img width="270" alt="image" src="https://github.com/user-attachments/assets/7f04ee75-a17e-4c02-8bd0-0648608e911b" />|


| **p.12** – Review screen | **p.12** - group selection | **p.12** - delete confirmation |
|---|---|---|
|<img width="270"  alt="image" src="https://github.com/user-attachments/assets/01c547ed-62f4-412c-be79-89cdffe6dba0" />|<img width="270"  alt="image" src="https://github.com/user-attachments/assets/71f3358f-4729-468c-bd28-6a3b35b5c1a4" />|<img width="270" alt="image" src="https://github.com/user-attachments/assets/bdcc77de-ee28-4351-a2fe-08e9766d2aa4" />|

| **p.13** – Preview screen for unsynced asset | **p.13** - delete confirmation |
|---|---|
|<img width="270" alt="image" src="https://github.com/user-attachments/assets/6a696218-f959-4188-b3bd-04e1d558a817" />|<img width="270" alt="image" src="https://github.com/user-attachments/assets/6625ce4e-1267-418c-913e-14a5df228d12" />|

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation if applicable
- [x] I have no unrelated changes in the PR.
- [x] I have confirmed that any new dependencies are strictly necessary.
- [x] I have written tests for new code (if applicable)
- [x] I have followed naming conventions/patterns in the surrounding code
- [x] All code in `src/services/` uses repositories implementations for database calls, filesystem operations, etc.
- [x] All code in `src/repositories/` is pretty basic/simple and does not have any immich specific logic (that belongs in `src/services/`)


...

Commit:
feature(mobile, beta, Android): handle remote asset trash/restore events and rescan media
- Handle move to trash and restore from trash for remote assets on Android
- Trigger MediaScannerConnection to rescan affected media files

Commit:
feature(mobile, beta, Android): fix rescan

Commit:
fix imports

Commit:
fix checking conditions

Commit:
refactor naming

Commit:
fix line breaks

Commit:
refactor code
rollback changes in BackgroundServicePlugin

Commit:
refactor code (use separate TrashService)

Commit:
refactor code

Commit:
parallelize restoreFromTrash calls with Future.wait
format trash.provider.dart

Commit:
try to re-format trash.provider.dart

Commit:
re-format trash.provider.dart

Commit:
rename TrashService to TrashSyncService to avoid duplicated names
revert changes in original trash.provider.dart

Commit:
create new table for out-of-sync records
add new experimental setting
update app setting page
fetch out-of-sync record if new setting
update timeline service for show out-of-sync assets
add out-of-sync review page (draft)
allow-deny actions (draft)

Commit:
remove redundant changes

Commit:
refactor(trash-sync): update services and review handling

- Refactor TrashSyncService and ActionService
- Respect isSyncApproved value in timeline.trashSyncReview
- Use i18n values instead of hardcoded strings
- Close review page when out-of-sync record count reaches 0

Commit:
close review page if all conflicts resolved

Commit:
refactor(trash-sync): rename setMoveToTrashDecision → resolveRemoteTrash, isApproved → allow

Commit:
sync remote restoration
refactor code

Commit:
show toast about exist out-of-sync changes
refactor outOfSyncCountProvider
create appSettingStreamProvider
ImmichToast: add warning mode, add onTap callback
refactor code

Commit:
refactor code
allow/deny buttons on asset preview (draft)

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	i18n/en.json
#	mobile/drift_schemas/main/drift_schema_v7.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/routing/router.dart
#	mobile/test/drift/main/generated/schema.dart
#	mobile/test/drift/main/generated/schema_v7.dart

Commit:
resolve merge conflicts

Commit:
refactor code (minor nitpicks)

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
revert changes related to toast
show warning mark on avatar instead
rework out-of-sync button

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	i18n/en.json
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/widgets/common/app_bar_dialog/app_bar_dialog.dart
#	mobile/lib/widgets/common/immich_sliver_app_bar.dart
#	mobile/lib/widgets/settings/advanced_settings.dart

Commit:
minor changes - remove debugPrint, use hardcoded label (temp)

Commit:
Merge branch 'feature/sync_assets_trashed_state' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/domain/services/trash_sync.service.dart

Commit:
resolve merge conflicts
refactor trash_sync service, action service
add label for Deny/Allow actions buttons

alextran1502 committed:
Merge branch 'main' into feature/sync_assets_trashed_state

Commit:
process restoreFromTrash sequentially instead of Future.wait

Commit:
group local assets by checksum before moving to trash
delete LocalAssetEntity records when moved to trash
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

Commit:
fix format

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

Commit:
use checksum for asset restoration
refactro code

Commit:
fix format

Commit:
sync trash only for backup-selected assets

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/test/domain/services/sync_stream_service_test.dart

Commit:
feat(db): add local_trashed_asset table and integrate with restoration flow
- Add new `local_trashed_asset` table to store metadata of trashed assets
- Save trashed asset info into `local_trashed_asset` before deletion
- Use `local_trashed_asset` as source for asset restoration
- Implement file restoration by `mediaId`

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v10.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v10.dart

Commit:
resolve merge conflicts

Commit:
fix index creating on migration

Commit:
Merge remote-tracking branch 'public/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/lib/infrastructure/repositories/local_asset.repository.dart

Commit:
rework trashed assets handling
- add new table trashed_local_asset
- mirror trashed assets data in trashed_local_asset.
- compute checksums for assets trashed out-of-app.
- restore assets present in trashed_local_asset and non-trashed in remote_asset.
- simplify moving-to-trash logic based on remote_asset events.

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/android/app/src/main/kotlin/app/alextran/immich/sync/Messages.g.kt
#	mobile/ios/Runner/Sync/Messages.g.swift
#	mobile/lib/domain/services/hash.service.dart
#	mobile/lib/domain/services/local_sync.service.dart
#	mobile/lib/platform/native_sync_api.g.dart
#	mobile/lib/providers/infrastructure/sync.provider.dart
#	mobile/pigeon/native_sync_api.dart
#	mobile/test/domain/services/hash_service_test.dart

Commit:
resolve merge conflicts
use updated approach for calculating checksums

Commit:
use CurrentPlatform instead _platform
fix mocks

Commit:
revert redundant changes

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
Include trashed items in getMediaChanges
Process trashed items delta during incremental sync

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v11.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v11.dart

Commit:
fix merge conflicts

Commit:
fix format

Commit:
trashed_local_asset table mirror of local_asset table structure
trashed_local_asset<->local_asset transfer data on move to trash or restore
refactor code

Commit:
refactor and format code

Commit:
refactor TrashedAsset model
fix missed data transfering

Commit:
refactor code
remove unused model

Commit:
fix label

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v12.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema.dart
#	mobile/test/drift/main/generated/schema_v12.dart

Commit:
fix merge conflicts

PeterOmbodi committed:
optimize, refactor code
remove redundant code and checking
getTrashedAssetsForAlbum for iOS
tests for hash trashed assets

PeterOmbodi committed:
format code

PeterOmbodi committed:
fix migration
fix tests

PeterOmbodi committed:
fix generated file

PeterOmbodi committed:
reuse exist checksums on trash data update
handle restoration errors
fix import

PeterOmbodi committed:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

PeterOmbodi committed:
format code

PeterOmbodi committed:
sync_stream.service depend on repos
refactor assets restoration
update dependencies in tests

PeterOmbodi committed:
remove trashed asset model
remove trash_sync.service
refactor DriftTrashedLocalAssetRepository, LocalSyncService

PeterOmbodi committed:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
rework fetching trashed assets data on native side
optimize handling trashed assets in local sync service
refactor code

PeterOmbodi committed:
update NativeSyncApi on iOS side
remove unused code

PeterOmbodi committed:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/ios/Podfile.lock

PeterOmbodi committed:
optimize sync trashed assets call in full sync mode
refactor code

PeterOmbodi committed:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

PeterOmbodi committed:
fix format

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
remove albumIds from getTrashedAssets params
fix upsert in trashed local asset repo
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/ios/Runner/Sync/MessagesImpl.swift

Commit:
fix getTrashedAssets params

Commit:
fix(trash-sync): clean up NativeSyncApiImplBase and correct applyDelta

Commit:
refactor(trash-sync): optimize performance and fix minor issues

Commit:
refactor(trash-sync): add missed index

Commit:
feat(trash-sync): remove sinceLastCheckpoint param from getTrashedAssets

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/ios/Runner/Sync/MessagesImpl.swift

Commit:
fix(trash-sync): fix target table

Commit:
fix(trash-sync): remove unused extension

Commit:
fix(trash-sync): remove unused code

Commit:
fix(trash-sync): refactor code

PeterOmbodi committed:
Merge branch 'immich-app:main' into feature/sync_assets_trashed_state

Commit:
Merge remote-tracking branch 'origin/feature/sync_assets_trashed_state' into feature/sync_assets_trashed_state

Commit:
fix(trash-sync): reformat file

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/lib/constants/constants.dart

Commit:
Merge branch 'refs/heads/feature/sync_assets_trashed_state' into feature/out-of-sync_assets_v2

# Conflicts:
#	i18n/en.json
#	mobile/drift_schemas/main/drift_schema_v9.json
#	mobile/lib/domain/models/store.model.dart
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/domain/services/trash_sync.service.dart
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/providers/infrastructure/trash_sync.provider.dart
#	mobile/lib/widgets/common/app_bar_dialog/app_bar_dialog.dart
#	mobile/lib/widgets/common/immich_sliver_app_bar.dart
#	mobile/test/drift/main/generated/schema.dart
#	mobile/test/drift/main/generated/schema_v9.dart

Commit:
fix(out-of-sync_trash): resolve merge conflicts (draft)

Commit:
fix(trash_sync): refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
fix(trash_sync): improve moving to trash

Commit:
refactor(trash_sync): integrate MANAGE_MEDIA permission request into login flow and advanced settings

Commit:
refactor(trash_sync): add additional checking for experimental trash sync flag and MANAGE_MEDIA permission.

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/lib/platform/native_sync_api.g.dart

Commit:
refactor(trash_sync): resolve merge conflicts

Commit:
refactor(trash_sync): fix format

PeterOmbodi committed:
Merge branch 'main' into feature/sync_assets_trashed_state

PeterOmbodi committed:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

# Conflicts:
#	mobile/lib/widgets/forms/login/login_form.dart
#	mobile/test/infrastructure/repository.mock.dart

PeterOmbodi committed:
resolve merge conflicts
add await for alert dialog
add missed request

Commit:
Merge branch 'feature/sync_assets_trashed_state' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/domain/services/local_sync.service.dart
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/domain/services/timeline.service.dart
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/widgets/common/app_bar_dialog/app_bar_dialog.dart
#	mobile/lib/widgets/settings/advanced_settings.dart

Commit:
feat(trash_sync_review): resolve merge conflicts
review very draft implementation
settings screen draft implementation

Commit:
feat(trash_sync_review): rework MANAGE_MEDIA info widget
refactor setting getter
fix tests

Commit:
refactor(trash_sync): rework MANAGE_MEDIA info widget
show rationale text in permission request alert dialog
refactor setting getter

Commit:
Merge remote-tracking branch 'origin/main' into feature/sync_assets_trashed_state

Commit:
Merge branch 'feature/sync_assets_trashed_state' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/domain/services/local_sync.service.dart
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/widgets/forms/login/login_form.dart
#	mobile/lib/widgets/settings/advanced_settings.dart

Commit:
fix(trash_sync_review): resolve merge conflicts

PeterOmbodi committed:
fix(trash_sync): format file

Commit:
fix(trash_sync_review): extend TrashSync table and model by actionType
cleaun-up trash sync table
local sync: trash sync review flow (draft)
advanced settings screen experimental widget

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
fix(trash_sync_review): modify TrashSync table structure
refine sync trash flow (draft. v2)

Commit:
refactor(trash_sync_review): refine sync trash flow (draft. v3)
use TabView for separate trashed and restored assets
create restored assets timeline
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/constants/constants.dart

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): remove code related to restoration review
use auto restoration logic
add alert dialog on move to trash
rework deleteAlreadySynced
update related labels
cleanup TrashSyncService

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/domain/services/local_sync.service.dart
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/lib/infrastructure/repositories/local_asset.repository.dart
#	mobile/lib/infrastructure/repositories/trashed_local_asset.repository.dart
#	mobile/lib/providers/infrastructure/sync.provider.dart
#	mobile/lib/providers/infrastructure/trash_sync.provider.dart
#	mobile/lib/widgets/forms/login/login_form.dart
#	mobile/lib/widgets/settings/advanced_settings.dart
#	mobile/lib/widgets/settings/beta_sync_settings/sync_status_and_actions.dart
#	mobile/test/domain/services/sync_stream_service_test.dart
#	mobile/test/drift/main/generated/schema.dart
#	mobile/test/infrastructure/repository.mock.dart

Commit:
feat(trash_sync_review): resolve merge conflicts

Commit:
feat(trash_sync_review): fix group by date

Commit:
feat(trash_sync_review): group sync trash timeline by checksum
include in merged assets deleted assets presented as local
show next asset on users decision
use diff design for action buttons
hide some widgets in sync mode

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): polish review flow
path deletedAt to LocalAsset
fix mergedBucket date

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart

Commit:
feat(trash_sync_review): cleunup trash_sync repo
add remoteDeletedAt in LocalAsset
optimize trash_sync table
use remote.deletedAt for resolve conflicts
refactor code

Commit:
feat(trash_sync_review): fix migration
fix watchIsApprovalPending
refactor code
show out-of-sync status (experimental)

Commit:
feat(trash_sync_review): use radioButtons instead toggles in AdvancedSettings
add subtitle to SettingsRadioGroup
more clear labels
refactor code

Commit:
feat(trash_sync_review): fix raw sql

Commit:
feat(trash_sync_review): remove accidental changes

PeterOmbodi committed:
fix(trash_sync_review): fix format

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): fix tests
fix format

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): fix review in time-line (fetch, update on review)
fix unSync icon
fix format
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): remove unnecessary column from table structure
refactor related code

Commit:
feat(trash_sync_review): fix format

Commit:
feat(trash_sync_review): remove unused methods and params
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): add review actions on asset details bottom-sheet
refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(trash_sync_review): increase timeline update delay

Commit:
Merge remote-tracking branch 'refs/remotes/origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v14.json
#	mobile/lib/domain/models/asset/local_asset.model.dart
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v14.dart

Commit:
feat(trash_sync_review): resolve merge conflicts

Commit:
feat(trash_sync_review): fix for edge case

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/top_app_bar.widget.dart

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/top_app_bar.widget.dart

Commit:
feat(trash_sync_review): resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/presentation/widgets/asset_viewer/bottom_sheet.widget.dart
#	mobile/lib/utils/action_button.utils.dart

Commit:
fix(trash_sync_review): resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
feat(mobile): Refactor trash-sync action buttons to use result callbacks and update viewer refresh/toast handling
Fix controls flickering during asset/timeline updates
Update keep result text

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	i18n/en.json
#	mobile/lib/infrastructure/repositories/local_asset.repository.dart

Commit:
fix(mobile): Set warning colors via ThemeData colorScheme.tertiary
Use colorScheme.tertiary for warning button color

Commit:
fix(mobile): improve deleting dialog
fix label style

Commit:
fix(mobile): fix&refactor logic in resolveRemoteTrash

Commit:
fix(mobile): remove doubled call

Commit:
test(mobile): add trash review flow tests

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
refactor(mobile): split PR changes, remove code not related to showing review timeline

Commit:
Merge remote-tracking branch 'public/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v15.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v15.dart

Commit:
fix(mobile): resolve merge conflicts

Commit:
fix(mobile): regenerate files

Commit:
fix(mobile): fix format

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v16.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v16.dart

Commit:
fix(mobile): resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v17.json
#	mobile/lib/domain/models/asset/local_asset.model.dart
#	mobile/lib/domain/models/asset/remote_asset.model.dart
#	mobile/lib/domain/services/sync_stream.service.dart
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/lib/widgets/settings/beta_sync_settings/sync_status_and_actions.dart
#	mobile/test/drift/main/generated/schema_v17.dart

Commit:
feature(mobile): Resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

shenlong-tanwen commented on mobile/lib/domain/services/trash_sync.service.dart:
```diff
@@ -0,0 +1,12 @@
+
+  Stream<int> watchPendingApprovalCount() => _trashSyncRepository.watchPendingApprovalCount();
+
+  Stream<bool> watchIsApprovalPending(String checksum) => _trashSyncRepository.watchIsApprovalPending(checksum);
```
Can we rename it to something like `watchIsApprovalPendingForAsset` or something that makes it much more obvious that this is asset specific and the one above is the total count of assets that are pending approval?

PeterOmbodi commented on mobile/lib/infrastructure/repositories/local_asset.repository.dart:
```diff
@@ -123,7 +123,7 @@ class DriftLocalAssetRepository extends DriftDatabaseRepository {
         final albumId = row.readTable(_db.localAlbumAssetEntity).albumId;
         final assetData = row.readTable(_db.localAssetEntity);
-        final asset = assetData.toDto();
+        final asset = assetData.toDto().copyWith(deletedAt: trashedAssetsMap[assetData.checksum]);
```
The response from `localAssetsToTrash` is then passed further in the code to
`await _trashSyncRepository.upsertReviewCandidates(itemsToReview)`,
where it is used in the condition
```
if (existing == null ||
    (existing.isSyncApproved == false &&
     item.deletedAt!.isAfter(existing.updatedAt)))
```
to handle the case of an asset being restored and deleted again.

Commit:
refactor(mobile): address review feedback

Commit:
fix(mobile): regenerate db.repository.drift.dart

Commit:
fix(mobile): address review feedback, p#2
use deleteOutdatedThrottled instead deleteOutdated
refactor var names

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	i18n/en.json
#	mobile/lib/domain/models/store.model.dart
#	mobile/lib/widgets/common/app_bar_dialog/app_bar_dialog.dart
#	mobile/lib/widgets/settings/advanced_settings.dart

Commit:
fix(mobile): address review feedback, p#3
remove code related to review logic

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
fix(mobile): fix tests

Commit:
refactor(mobile): refactor code

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v18.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/drift/main/generated/schema_v18.dart

Commit:
fix(mobile): resolve merge conflicts

Commit:
refactor(mobile): introduce RemoteDeletedLocalAsset model
- move remote deletion timestamp off LocalAsset
- use RemoteDeletedLocalAsset in trash/sync repositories and services
- update drift schema artifacts and related tests

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
fix(mobile): format code

Commit:
fix(mobile): format code

Commit:
refactor(mobile): move getToTrash to local asset repo and update tests

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/domain/models/store.model.dart
#	mobile/test/domain/services/sync_stream_service_test.dart

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/widgets/common/immich_sliver_app_bar.dart

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v19.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.drift.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/lib/widgets/common/app_bar_dialog/app_bar_dialog.dart
#	mobile/test/drift/main/generated/schema_v19.dart

Commit:
chore(mobile): resolve DB merge conflicts, bump version, and regenerate schemas

Commit:
chore(mobile): regenerate db.repository.drift.dart

shenlong-tanwen commented on mobile/lib/domain/models/asset/remote_deleted_local_asset.dart:
```diff
@@ -0,0 +1,8 @@
+import 'package:immich_mobile/domain/models/asset/base_asset.model.dart';
+
+class RemoteDeletedLocalAsset {
```
Feels weird to have a class to do this. Can we use named records instead?

Commit:
chore(mobile): address review comments #1

Commit:
refactor(mobile): remove code related to deleteOutdated

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

Commit:
chore(mobile): fix format

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/lib/presentation/widgets/asset_viewer/bottom_bar.widget.dart
#	mobile/lib/presentation/widgets/asset_viewer/viewer_top_app_bar.widget.dart

Commit:
chore(mobile): resolve merge conflicts

Commit:
Merge remote-tracking branch 'origin/main' into feature/out-of-sync_assets_v2

# Conflicts:
#	mobile/drift_schemas/main/drift_schema_v20.json
#	mobile/lib/infrastructure/repositories/db.repository.dart
#	mobile/lib/infrastructure/repositories/db.repository.steps.dart
#	mobile/test/domain/services/sync_stream_service_test.dart
#	mobile/test/drift/main/generated/schema_v20.dart

Commit:
refactor(mobile): resolve merge conflicts
refactor ActionButtonContext

Commit:
fix(mobile): fix test init
